src/crewai/memory/storage/rag_storage.py

```diff
@@ -81,6 +85,29 @@ def _sanitize_role(self, role: str) -> str:
         """
         return role.replace("\n", "").replace(" ", "_").replace("/", "_")
 
+    def _build_storage_file_name(self, type: str, file_name: str) -> str:
+        """
+        Ensures file name does not exceed max allowed by OS
+        """
+        base_path = f"{db_storage_path()}/{type}"
+
+        try:
+            # Returns platform-dependent max length for a file name
+            max_length = os.pathconf(base_path, "PC_NAME_MAX")
```

Comment:
os.pathconf didnt work for me.

Alternative
```python
 # Use a conservative default max length that works across most systems
        max_length = 255  # Standard length supported by most modern filesystems

        # Trim if necessary
        if len(file_name) > max_length:
            logging.warning(
                f"Trimming file name from {len(file_name)} to {max_length} characters."
            )
            file_name = file_name[:max_length]
```
<img width="1618" alt="Screenshot 2024-12-03 at 2 31 10 PM" src="https://github.com/user-attachments/assets/677586ad-2df9-4709-98f0-64059c25281e">

Reply:
I wonder if it's because `{type}` is not part of the path when it was first created from inside `db_storage_path()`. Conservative 255 is probably fine unless `db_storage_path()` can be updated to take optional value of `type` for when it calls `appdirs.mkdir`.

Reply:
Thats right. I'd recommend just doing:
255 as most filesystems support that.

Reply:
If you can knock this out with maybe using the suggested above we can get this merged in

Reply:
@lorenzejay Resolved!